src/test/java/discovery/controller/DiscoveryControllerTestSpring.java

```diff
@@ -0,0 +1,122 @@
+package discovery.controller;
+
+
+import discovery.Application;
+import discovery.configuration.DiscoveryConfig;
+import discovery.model.Discover;
+import org.junit.Before;
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.MethodSorters;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@WebAppConfiguration
+@ContextConfiguration(classes = Application.class)
+public class DiscoveryControllerTestSpring {
```

Comment:
good job, looking ahead, you could delete/disable this class, so you don't have to refactor too many tests when you changed behaviour